livepeer/go-livepeer
# Add foundation for E2E Tests + First E2E Scenario: Register Orchestrator

**What does this pull request do? Explain your changes. (required)**
Create first E2E Scenario which registers an orchestrator using local chain.

How to run the test:
1. You need to have Docker running
2. (Mac M1) Update `Image` in `e2e.go` to `darkdragon/geth-with-livepeer-protocol:streamflow`
3. Fire `./test_e2e.sh`

For more details about next steps, please check [E2E Tests Epic](https://github.com/livepeer/go-livepeer/issues/2297)

**Specific updates (required)**
- Extract `devtool` package from `devtool.go` to make the devtool code reusable
- Extract `starter` package from `livepeer.go` to make the livepeer code reusable
- Create `e2e.go` with the reusable utils for E2E Tests
- Create `register_orchestrator_test.go` with the first E2E Test scenario
- Create/Update `test.sh`/`test_e2e.sh`

**How did you test each of these updates (required)**

- `./test-e2e.sh`
- check if nothing is broken with devtool
- check if nothing is broken with livepeer


**Does this pull request close any open issues?**
fix #2317


**Checklist:**
